modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/source/poll/PollingSourceWrapper.java

```diff
@@ -204,6 +216,21 @@ private int compareWatermarks(Serializable w1, Serializable w2, Comparator compa
     return comparator.compare(w1, w2);
   }
 
+  @Override
+  public RestartContext getRestartContext() {
+    restarting.set(true);
```

Comment:
a get method should never have side effects

Reply:
The source need to know that a restart is coming so that it does not kill the scheduler. Maybe we should change the name of the method?

Reply:
you can change the mehthod name, but still, is this the right place for doing this?